File: backend/lib/L2_data/schema/goals/remote_tracker.py
#  Copyright (c) 2022. Alexandr Moroz

from abc import ABC
from typing import Optional

from pydantic import HttpUrl

from ..base_schema import PKGetable, PKUpsertable, WorkspaceBounded
from .remote_tracker_type import RemoteTrackerTypeSchemaGet


class _RemoteTrackerSchema(WorkspaceBounded, ABC):
    url: HttpUrl
    login_key: str
    password: Optional[str]
    description: Optional[str]


class RemoteTrackerSchemaGet(_RemoteTrackerSchema, PKGetable):
    type: RemoteTrackerTypeSchemaGet


class RemoteTrackerSchemaUpsert(_RemoteTrackerSchema, PKUpsertable):
    remote_tracker_type_id: Optional[int]

File: lib/L1_domain/api_schema/remote_tracker.dart
// Copyright (c) 2022. Alexandr Moroz

import 'base_upsert.dart';

class RemoteTrackerUpsert extends BaseUpsert {
  RemoteTrackerUpsert({
    required int? id,
    required this.typeId,
    required this.url,
    required this.loginKey,
    required this.password,
    required this.description,
    required this.workspaceId,
  }) : super(id: id);

  final int typeId;
  final String url;
  final String loginKey;
  final String password;
  final String? description;

  final int workspaceId;
}
